### PR TITLE
support for termination of partitioned topic

### LINKED ISSUE #5905: [pulsar-broker][REST API] Support `terminate` all partitions of a partitioned topic
**Is your feature request related to a problem? Please describe.**

Pulsar supports `terminate` a non-partitioned-topic or a partition of a partitioned topic. If there has a partitioned topic with too many partitions, users need to get all partitions and `terminate` them one by one. We need to support `terminate` all partition of a partitioned topic.

**Describe the solution you'd like**

Support `terminate` all partition of a partitioned topic.

The implementation is similar to this: https://github.com/apache/pulsar/blob/bdb1f4cb3d1df27957a1472e02d7a22f229aaef8/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java#L890
We need to get all partition of a partitioned topic if the topic is a partitioned topic and send `terminate` requests and waiting for them back.

**Additional context**

We need to add an integration test or unit test for verifying this change.

There has a similar unit test about REST API: https://github.com/apache/pulsar/blob/bdb1f4cb3d1df27957a1472e02d7a22f229aaef8/pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/PersistentTopicsTest.java#L136

### PR DESCRIPTION
**Pull request by antimpatel**
Fixes #5905 

### Motivation
Pulsar supports terminate a non-partitioned-topic or a partition of a partitioned topic. If there has a partitioned topic with too many partitions, users need to get all partitions and terminate them one by one. We need to support terminate all partition of a partitioned topic.

### Modifications
Added API to support terminate all partitioned topic in PersistenceTopicBase. and added test case for the same.

### Verifying this change
added test case and verified.

### CODE REVIEW

**Comment by antimpatel** [comment]
@zymap Please check the PR, against the mentioned issue.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/PersistentTopicsTest.java [hunk 1]
```diff
@@ -200,6 +203,76 @@ public void testGetSubscriptions() {
         Assert.assertEquals(responseCaptor.getValue().getStatus(), Response.Status.NO_CONTENT.getStatusCode());
     }
 
+    @Test
+    public void testTerminatePartitionedTopic() {
+        String testLocalTopicName = "topic-not-found";
```
Comment (jiazhai) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/PersistentTopicsTest.java]: Seems most of the test content is duplicate with above `testGetSubscriptions()`,  Could we refine this test case to focus on topic Terminate?

**Comment by jiazhai** [comment]
Thanks @antimpatel for the help. left a comments related to the test.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/PersistentTopicsTest.java (hunk 1 above)
Comment (ptlanu22) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/PersistentTopicsTest.java]: Simplified only for termination of topic.

**Comment by codelipenghui** [comment]
@antimpatel There are some CI tests failed, could you please merge the master branch? since there are flaky tests are fixed in master branch.

**Pull request merged**